modules/ml/src/rtrees.cpp

```diff
@@ -425,6 +425,7 @@ class DTreesImplForRTrees CV_FINAL : public DTreesImpl
     vector<float> varImportance;
     vector<int> allVars, activeVars;
     RNG rng;
+    RNG initRng = RNG((uint64) - 1);
```

Comment:
What is about this: https://github.com/alalek/opencv/commit/pr16251_r ?
Lets re-use global `theRNG()` (it allows to save / restore RNG state too).

```
cv::theRNG() = RNG(your train seed);
... call train() ...
```

Reply:
Personally, I'd be ok with this. But it would break the default behavior of `RTrees`.
People might be relying on deterministic `RTrees` results.
Honestly, I was actually surprised that OpenCV does this ^^.